File: src/AutomatedInvoiceGenerator/Models/ServiceItem/ServiceItemsSet.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace AutomatedInvoiceGenerator.Models
{
    public class ServiceItemsSet : IAuditable
    {
        public int Id { get; set; }

        [Timestamp]
        public byte[] RowVersion { get; set; }


        // Relationships
        public int CustomerId { get; set; }
        [Display(Name = "Kontrahent:")]
        public virtual Customer Customer { get; set; }

        [Display(Name = "Faktury:")]
        public virtual ICollection<Invoice> Invoices { get; set; }

        [Display(Name = "Usługi jednorazowe:")]
        public virtual ICollection<OneTimeServiceItem> OneTimeServiceItems { get; set; }

        [Display(Name = "Usługi abonamentowe:")]
        public virtual ICollection<SubscriptionServiceItem> SubscriptionServiceItems { get; set; }
    }
}

File: src/FrontendApp/src/app/service-items-sets/service-items-set.ts
import { OneTimeServiceItem } from '../one-time-service-items/one-time-service-item';
import { SubscriptionServiceItem } from '../subscription-service-items/subscription-service-item';

export class ServiceItemsSet {
    id: number;
    name: string;
    rowVersion: string;
    customerId: number;

    oneTimeServiceItems: OneTimeServiceItem[];
    subscriptionServiceItems: SubscriptionServiceItem[];
}
